src/amocrm/services/orders/order_pusher.py

```diff
@@ -0,0 +1,80 @@
+from dataclasses import dataclass
+
+from amocrm.exceptions import AmoCRMServiceException
+from amocrm.models import AmoCRMOrderLead
+from app.services import BaseService
+from orders.models import Order
+
+
+class AmoCRMOrderPusherException(AmoCRMServiceException):
+    """Raises when it's impossible to push order to amocrm"""
+
+
+@dataclass
+class AmoCRMOrderPusher(BaseService):
+    """
+    Push given order to amocrm
+    """
+
+    order: Order
+
+    def act(self) -> None:
+        if not self.order_must_be_pushed():
+            return
+
+        if self.order.paid is not None or self.order.unpaid is not None:
+            self.push_order()
+        else:
+            self.push_lead()
+
+    def push_order(self) -> None:
+        from amocrm.tasks import push_existing_order_to_amocrm
+
+        existing_lead = self.get_lead()
+        if existing_lead is None:
+            raise AmoCRMOrderPusherException("Cannot push paid or unpaid order without existing lead")
+        if existing_lead.order != self.order:
+            self.link_existing_lead_to_new_order(existing_lead=existing_lead)
+
+        push_existing_order_to_amocrm.delay(order_id=self.order.id)
+
+    def push_lead(self) -> None:
+        from amocrm.tasks import create_amocrm_lead
+        from amocrm.tasks import update_amocrm_lead
+
+        existing_lead = self.get_lead()
+        if existing_lead is not None:
```

Comment:
(Вкусовщина). Лучше поменять местами if и else, написав более короткую ветвь выше. Так код будет читаться легче:
```suggestion
        if existing_lead is None:
          create_lead()
        else:
          ...
```